components/list/list-item-expand-collapse-mixin.js

```diff
@@ -0,0 +1,161 @@
+import { EventSubscriberController } from '../../controllers/subscriber/subscriberControllers.js';
+
+const dragIntervalDelay = 100;
+const dragHoverDropTime = 1000;
+
+export const ListItemExpandCollapseMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			/**
+			 * Whether to show the expand collapse toggle
+			 * @type {boolean}
+			 */
+			expandable: { type: Boolean },
+			/**
+			 * Default state for expand collapse toggle - if not set, collapsed will be the default state
+			 * @type {boolean}
+			 */
+			expanded: { type: Boolean, reflect: true },
+			_siblingHasNestedItems: { state: true },
+			_renderExpandCollapseSlot: { type: Boolean, reflect: true, attribute: '_render-expand-collapse-slot' },
+			_showNestedLoadingSpinner: {  state: true }
+		};
+	}
+
+	static get styles() {
+		const styles = [ css`
+			:host {
+				--d2l-expand-collapse-slot-transition-duration: 0.3s;
+			}
+			.d2l-list-expand-collapse {
+				transition: width var(--d2l-expand-collapse-slot-transition-duration) cubic-bezier(0, 0.7, 0.5, 1);
+				width: 0;
+			}
+			:host([dir="rtl"][_render-expand-collapse-slot]) .d2l-list-expand-collapse {
+				padding: 0.4rem 0 0 0.3rem;
+			}
+			.d2l-list-expand-collapse d2l-button-icon {
+				--d2l-button-icon-min-height: 1.2rem;
+				--d2l-button-icon-min-width: 1.2rem;
+			}
+			.d2l-list-expand-collapse:hover d2l-button-icon {
+				background-color: var(--d2l-button-icon-background-color-hover);
+				border-radius: var(--d2l-button-icon-border-radius);
+			}
+			:host([_render-expand-collapse-slot]) .d2l-list-expand-collapse {
+				padding: 0.4rem 0.3rem 0 0;
+				width: 1.2rem;
+			}
+			.d2l-list-expand-collapse-action {
+				cursor: pointer;
+				display: block;
+				height: 100%;
+			}
+			.d2l-list-nested-loading {
+				display: flex;
+				justify-content: center;
+				padding: 0.4rem;
+			}
+		` ];
+
+		super.styles && styles.unshift(super.styles);
+		return styles;
+	}
+
+	constructor() {
+		super();
+		this._siblingHasNestedItems = false;
+		this._renderExpandCollapseSlot = false;
+		this._showNestedLoadingSpinner = false;
+		this._parentChildUpdateSubscription = new EventSubscriberController(this, {}, { eventName: 'd2l-list-child-status' });
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		// mixin requires key for events
+		if (!this.key && this.expandable) {
+			console.warn('ListItemExpandCollapseMixin requires a key.');
+			this.expandable = false;
+		}
+	}
+
+	updated(changedProperties) {
+		if (changedProperties.has('_siblingHasNestedItems') || changedProperties.has('expandable')) {
+			this._renderExpandCollapseSlot = this.expandable || this._siblingHasNestedItems;
+		}
+		if (changedProperties.has('_draggingOver') && this._draggingOver && this.dropNested && !this.expanded && this.expandable) {
+			let elapsedHoverTime = 0;
+			let dragIntervalId = null;
+			const watchDraggingOver = () => {
+				if (elapsedHoverTime === dragHoverDropTime) {
```

Comment:
Nit: this relies on the `dragHoverDropTime` being evenly divisible by the interval, which is true as configured, but if either were ever changed there is a reasonable chance it would no longer be, and this condition might never be met. Suggest change this to `>=`.